recipes/elfutils/all/conanfile.py

```diff
@@ -0,0 +1,161 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+import os
+import glob
+
+
+class ElfutilsConan(ConanFile):
+    name = "elfutils"
+    description = "A dwarf, dwfl and dwelf functions to read DWARF, find separate debuginfo, symbols and inspect process state."
+    homepage = "https://sourceware.org/elfutils"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("conan", "elfutils", "libelf", "libdw", "libasm")
+    exports = ["patches/elfutils-*.patch"]
+    license = ["GPL-1.0-or-later", "LGPL-3.0-or-later", "GPL-3.0-or-later"]
+    
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "shared": True,
+        "fPIC": True
+    }
+    requires = (
+        "bzip2/1.0.6",
+        "zlib/1.2.11",
+        "xz_utils/5.2.4"
+    )
+
+    _autotools = None
+    _source_subfolder = "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "elfutils" + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_autotools(self):
+        if not self._autotools:
+            args = ['--enable-deterministic-archives', '--enable-silent-rules', '--with-zlib', '--with-bzlib', '--with-lzma']
+            self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+            self._autotools.configure(configure_dir=self._source_subfolder, args=args)
+        return self._autotools
+
+    def build_id(self):
+        self.info_build.options.shared = "Any"
+    
+    def build(self):
+        tools.patch(**self.conan_data["patches"][self.version])
+        autotools = self._configure_autotools()
+        autotools.make()
+    
+    def package(self):
+        self.copy(pattern="COPYING*", dst="licenses", src=self._source_subfolder)
+        autotools = self._configure_autotools()
+        autotools.install()
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
```

Comment:
That's a good question. Should the components name be named like in pkg-config for simplicity? Or something like this `self.cpp_info.components["elf"].names["pkg_info"] = "libelf"` be prefered?

Regarding the original pkg-config generated files, only `libdw.pc` and `libelf.pc` are generated with the names `libdw` and `libelf` respectively.

Reply:
The names of the components matter in the sense that they show up in generators as a default name.
I think you can simply create a `libdw` and a `libelf` component with the `pkg_config` names set accordingly.

cci has a `libelf` recipe. Won't this clash with the `libelf.pc` generated by this recipe?

Reply:
Yep if we rename the component name to `libelf` it could clash. With components one would expect to be able to say `elfutils::libelf` but for `pkg_config` I don't know what's the generator behavior. Without knowing how this recipy is going to be consumed is a bit difficult to reason about.
Maybe I should wait until the components feature is more mature? (https://github.com/conan-io/conan/issues/6717 https://github.com/conan-io/conan/issues/6718)